src/study-access/permission.ts

```diff
@@ -157,9 +157,6 @@ export function isUserApprovedForAction(
   datasetId: string,
   action: Action,
 ) {
```

Comment:
```suggestion
) {
  if (user.properties?.skipStudyPermissions) {
    return true;
  }
```
```

Reply:
(This is necessary because the "stubbed" permissions `{ type: 'external', perDataset: {} }` permit _no_ UI actions in _any_ study. Alternatively: in our group slack, @dmfalke had [an interesting suggestion](https://epvb.slack.com/archives/C0334MQA6NR/p1645041540553579?thread_ts=1645041280.039709&cid=C0334MQA6NR) of using JavaScript `Proxy`s to make a permissions object which permits _all_ UI actions in _all_ studies.)

Reply:
Alternatively: in our group slack, @dmfalke had [an interesting suggestion](https://epvb.slack.com/archives/C0334MQA6NR/p1645041540553579?thread_ts=1645041280.039709&cid=C0334MQA6NR) of using JavaScript `Proxy`s to make a permissions object which permits _all_ UI actions in _all_ studies.

Reply:
If we did this, then I think we could simply get rid of all of these checks, _and_ the new user property.

Instead, we could change https://github.com/VEuPathDB/web-study-data-access/blob/48a3035ef564a065bec834c6926366f68745e475/src/shared/wrapWdkDependencies.ts#L18-L31 such that `studyAccessApiUrl` is optional, and if not defined, use a stubbed version of `StudyAccessApi` that includes this `Proxy`-based object.

What I like about this is that we don't have to introduce the hacky `skipStudyPermissions` property.

Reply:
I'm quite fond of the `Proxy`-based solution, and doing the stubbing at the level of `wrapWdkDependencies`, as @dmfalke suggests.